drivers/flash/flash_shell.c

```diff
@@ -736,9 +736,14 @@ static void device_name_get(size_t idx, struct shell_static_entry *entry);
 
 SHELL_DYNAMIC_CMD_CREATE(dsub_device_name, device_name_get);
 
+static bool device_is_flash(const struct device *dev)
+{
+	return DEVICE_API_IS(flash, dev);
+}
+
```

Comment:
I wonder whether this should be defined in the Flash API, as a "can I serve this?' utility function.

Reply:
yeah the idea was that this could be checked in the API calls, not sure where we left there, @pdgendt?